File: backend/src/main/java/com/coffeediary/dto/DiaryEntryRequest.java
package com.coffeediary.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DiaryEntryRequest {

    @NotNull(message = "Date/time is required")
    private LocalDateTime dateTime;

    private Long sieveId;

    @Builder.Default
    private Integer temperature = 93;

    private Long coffeeId;

    private Double grindSize;

    private Double inputWeight;

    private Double outputWeight;

    private Integer timeSeconds;

    @Min(value = 1, message = "Rating must be between 1 and 5")
    @Max(value = 5, message = "Rating must be between 1 and 5")
    private Integer rating;

    private String notes;
}

File: ios/CoffeeDiary/CoffeeDiary/Models/DiaryEntryRequest.swift
import Foundation

struct DiaryEntryRequest: Encodable {
    var dateTime: String
    var sieveId: Int64?
    var temperature: Int
    var coffeeId: Int64?
    var grindSize: Double?
    var inputWeight: Double?
    var outputWeight: Double?
    var timeSeconds: Int?
    var rating: Int?
    var notes: String?

    init(from entry: DiaryEntry? = nil) {
        if let entry {
            dateTime = entry.dateTime
            sieveId = entry.sieveId
            temperature = entry.temperature
            coffeeId = entry.coffeeId
            grindSize = entry.grindSize
            inputWeight = entry.inputWeight
            outputWeight = entry.outputWeight
            timeSeconds = entry.timeSeconds
            rating = entry.rating
            notes = entry.notes
        } else {
            dateTime = DiaryEntry.dateFormatter.string(from: Date())
            temperature = 93
        }
    }
}
